bin/i18n/utils/sync_out_base.rb

```diff
@@ -0,0 +1,55 @@
+require_relative '../i18n_script_utils'
+
+module I18n
+  module Utils
+    class SyncOutBase
+      class << self
+        def perform
+          new.send(:execute)
+        end
+
+        protected
+
+        def process(&block)
+          @process_block = block
+        end
+
+        private
+
+        def process_block
+          @process_block ||= proc {raise NotImplementedError}
+        end
+      end
+
+      protected
+
+      def languages
+        @languages ||= PegasusLanguages.all
```

Comment:
We also have DashboardLanguages. Do we only need pegasus languages?

Reply:
We use only `PegasusLanguage` in `sync-out`:
https://github.com/code-dot-org/code-dot-org/blob/4866b1745f50e74942b06823d60099d0da8ac014/bin/i18n/sync-out.rb#L75-L78

Reply:
I will add this as a future work. if Dashboard and Pegasus split, we may need to change this.